unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/IdleChecker.cs

```diff
@@ -0,0 +1,98 @@
+using UnityEngine;
+
+namespace DCL
+{
+    public interface IIdleChecker
+    {
+        void Initialize();
+        void Dispose();
+        void SetMaxTime(int time);
+        int GetMaxTime();
+        bool isIdle();
+        void Update();
+        
+        delegate void ChangeStatus(bool isIdle);
+        void Subscribe(ChangeStatus callback);
+        void Unsubscribe(ChangeStatus callback);
+    }
+    
+    public class IdleChecker : IIdleChecker
+    {
+        Vector3 lastMouseCoordinate = Vector3.zero;
+        private int maxTime = 60;
+        private float lastActivityTime = 0.0f;
+        private bool idle = false;
+        public event IIdleChecker.ChangeStatus OnChangeStatus;
+
+        public void Subscribe(IIdleChecker.ChangeStatus callback)
+        {
+            OnChangeStatus += callback;
+        }
+
+        public void Unsubscribe(IIdleChecker.ChangeStatus callback)
+        {
+            OnChangeStatus -= callback;
+        }
+
+        public void Initialize()
+        {
+            lastActivityTime = Time.time;
+        }
+
+        public void Dispose()
+        {
+            
+        }
+        
+        public void SetMaxTime(int time)
+        {
+            maxTime = time;
+        }
+
+        public int GetMaxTime()
+        {
+            return maxTime;
+        }
+
+        public void Update()
+        {
+            Vector3 mouseDelta = Input.mousePosition - lastMouseCoordinate;
+            
+            bool mouseMoved = mouseDelta.x < 0;
```

Comment:
If the mouse is moved only to the right, the `mouseMoved` bool will remain as false. Also, comparing floats against zero is bad practice due to precision inconsistencies that can arise.

I propose something like:
`mouseMoved = mouseDelta.sqrMagnitude > 0.0001f`

Reply:
nice catch